runtime-light/stdlib/zlib/zlib-functions.cpp

```diff
@@ -115,3 +131,151 @@ std::optional<string> decode(std::span<const char> data, int64_t encoding) noexc
 }
 
 } // namespace kphp::zlib
+
+class_instance<C$DeflateContext> f$deflate_init(int64_t encoding, const array<mixed>& options) noexcept {
+  int32_t level{-1};
+  int32_t memory{8};
+  int32_t window{15};
+  auto strategy{Z_DEFAULT_STRATEGY};
+  auto extract_int_option{[](int32_t lbound, int32_t ubound, const array_iterator<const mixed>& option, int32_t& dst) noexcept {
+    const mixed& value = option.get_value();
+    if (value.is_int() && value.as_int() >= lbound && value.as_int() <= ubound) {
+      dst = value.as_int();
+      return true;
+    } else {
+      kphp::log::warning("deflate_init() : option {} should be number between {}..{}", option.get_string_key().c_str(), lbound, ubound);
+      return false;
+    }
+  }};
+  switch (encoding) {
+  case kphp::zlib::ENCODING_RAW:
+  case kphp::zlib::ENCODING_DEFLATE:
+  case kphp::zlib::ENCODING_GZIP:
+    break;
+  default:
+    kphp::log::warning("deflate_init() : encoding should be one of ZLIB_ENCODING_RAW, ZLIB_ENCODING_DEFLATE, ZLIB_ENCODING_GZIP");
+    return {};
+  }
+  for (const auto& option : options) {
+    mixed value;
+    if (!option.is_string_key()) {
+      kphp::log::warning("deflate_init() : unsupported option");
+      return {};
+    }
+    if (option.get_string_key() == string{"level"}) {
```

Comment:
It looks like we don't need to create temporary string in such checks

Reply:
fixed